embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/comms/delivery/EmbraceDeliveryService.kt

```diff
@@ -99,6 +100,17 @@ internal class EmbraceDeliveryService(
             val allSessions = cacheManager.getAllCachedSessionIds().filter {
                 it.sessionId != sessionIdTracker.getActiveSessionId()
             }
+
+            allSessions.map { it.sessionId }.forEach { sessionId ->
+                cacheManager.transformSession(sessionId = sessionId) { sessionMessage ->
+                    val spansToFail = sessionMessage.spanSnapshots?.map {
+                        it.toFailedSpan(sessionMessage.session.endTime ?: 0L)
+                    } ?: emptyList()
+                    val completedSpans = spansToFail + (sessionMessage.spans ?: emptyList())
+                    sessionMessage.copy(spans = completedSpans, spanSnapshots = emptyList())
```

Comment:
If `transformSession` is called more than once then it looks like this would result in duplicate spans in the payload

Reply:
Oooh, good call. I'll fix and write a test for it. Didn't think this could happen but definitely should protect against that. The good thing is the server will reject dupes but that's unnecesary

Reply:
Made it so that if a conversion was already done or if a completed span already exists for that spanId, we just drop the snapshot